cpp/include/cucim/filesystem/file_handle.h

```diff
@@ -70,6 +74,13 @@ struct EXPORT_VISIBLE CuCIMFileHandle : public std::enable_shared_from_this<CuCI
             deleter(this);
             deleter = nullptr;
         }
+
+        if (own_fd)
```

Comment:
Should we check `fd` too? Is there any risk it is undefined?

Reply:
Thanks @jakirkham for the feedback! 
Will update to make sure to check fd (`fd >= 0`).